ChaosMod/Effects/db/Player/PlayerIllegalInnocence.cpp

```diff
@@ -0,0 +1,140 @@
+/*
+	Effect by ShySkream
+	
+	uses modified code from "pacifist", "need for speed", and "never wanted".
+*/
+
+#include <stdafx.h>
+
+#pragma region variable declarations
+#define WAIT_TIME 5000 // ms. before police are angry at your innocence
+
+static DWORD64 m_timeReserve;
+static DWORD64 m_lastTick = 0;
+
+static int lastPlayerHits;
+static Hash currentPedHash;
+
+static int lastWantedLevel = 0;
+#pragma endregion
+
+static void OnStart()
+{
+	#pragma region variable initializations
+	lastWantedLevel = PLAYER::GET_PLAYER_WANTED_LEVEL(PLAYER_ID());
+	m_lastTick = GET_GAME_TIMER();
+	m_timeReserve = WAIT_TIME;
+	currentPedHash = -1;
+	lastPlayerHits = -1;
+	#pragma endregion
+}
+
+static void OnTick()
+{
+	#pragma region variable updates
+	DWORD64 currentTick = GET_GAME_TIMER();
+	DWORD64 tickDelta = currentTick - m_lastTick;
+
+	Player player = PLAYER_ID();
+	int wantedLevel = PLAYER::GET_PLAYER_WANTED_LEVEL(player);
+
+	Hash playerHash = GET_ENTITY_MODEL(PLAYER_PED_ID());
+	#pragma endregion
+
+	#pragma region check if player ran over someone
+	if (GET_TIME_SINCE_PLAYER_HIT_PED(player) < 200 && GET_TIME_SINCE_PLAYER_HIT_PED(player) != -1) {
+		lastWantedLevel = 0;
+		SET_PLAYER_WANTED_LEVEL(player, lastWantedLevel, false);
```

Comment:
you should extract this and call it each time you use it

Reply:
This would make the code a little shorter, but it would also result in a lot of extra calls when nothing is happening.  Not that the extra calls would have much impact, I just like it to be efficient.

Reply:
> This would make the code a little shorter, but it would also result in a lot of extra calls when nothing is happening. Not that the extra calls would have much impact, I just like it to be efficient.

i think i selected the wrong bit, i mean just the lines that set the wanted level, not the whole if statement

Reply:
> i think i selected the wrong bit, i mean just the lines that set the wanted level, not the whole if statement

I was referring to the wanted level lines as well, yes.